kf_lib_data_ingest/validation/reporting/base.py

```diff
@@ -20,6 +20,21 @@
 
 
 class AbstractReportBuilder(ABC):
+    """
+    Abstract base class defines the required interface for all
+    subclass validation report builders
+
+    Subclasses must implement methods:
+    - `_build`: builds the report content in a particular format
+    - `_write_report`: writes the report content to disk.
+
+    The public `build` method will call `_validate_result_schema` to
+    ensure validation results adhere to the expected the structure/format
```

Comment:
```suggestion
    ensure validation results adhere to the expected structure/format
```

Reply:
This is my only comment regarding the code. For the overall structure, it looks good to me. But maybe @fiendish should review as well.